mongoengine/fields.py

```diff
@@ -2502,6 +2502,15 @@ def to_mongo(self, value):
         else:
             return pk
 
+    def to_python(self, value):
+        """Convert a MongoDB-compatible type to a Python type."""
+        if not self.dbref and not isinstance(
+            value, (DBRef, Document, EmbeddedDocument)
+        ):
+            collection = self.document_type._get_collection_name()
+            value = DBRef(collection, self.document_type.id.to_python(value))
+        return value
```

Comment:
document this as a breaking change in the CHANGELOG

Reply:
Done